icq-relayer/cmd/root.go

```diff
@@ -21,13 +22,17 @@ import (
 var (
 	homePath       string
 	overridenChain string
-	defaultHome    = os.ExpandEnv("$HOME/.icq")
 	appName        = "icq-relayer"
 )
 
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	defaultHome := filepath.Join(userHomeDir, ".icq")
```

Comment:
Let's not duplicate this functionality everywhere, let's extract this code into a helper

```go
var homeDirOnce sync.Once
var defaultHome string

func retrieveHomeDir() {
    homeDirOnce.Do(func() {
        userHomeDir, err := os.UserHomeDir()
        if err != nil {
            panic(err)
        }
        defaultHome = filepath.Join(userHomeDir, ".icq")
    })
    return defaultHome
}
```

and then use it allover instead of that duplication in icq-relayer/cmd/run.go

Reply:
thank you